### PR TITLE
Add upper_bound_ and is_symbolic_ to SizeNode

### PR DESCRIPTION
**Pull request by JackCaoG**
Refactor the Size Node a bit

https://github.com/pytorch/xla/pull/3954 needs to merge first.

### CODE REVIEW

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by vanbasten23** [comment]
Just curious, is this pr based off the other open pr https://github.com/pytorch/xla/pull/3954?

**Comment by JackCaoG** [comment]
> Just curious, is this pr based off the other open pr #3954?

yup... I am doing manual github pr stacking..

**Comment by JackCaoG** [comment]
I will merge this one after https://github.com/pytorch/xla/pull/3954 merged

File: torch_xla/csrc/ops/dynamic_ir.h [hunk 1]
```diff
@@ -38,40 +38,50 @@ namespace torch_xla {
 class SizeNode : public XlaNode, public torch::lazy::DimensionNode {
```
Comment (vanbasten23) [inline torch_xla/csrc/ops/dynamic_ir.h:38]: Irrelevant to this pr, but at this line 27 in this file, should it be `numel = x.size()[0] * x.size()[1]`? I couldn't find a `size()` for a tensor.

Comment (JackCaoG) [reply torch_xla/csrc/ops/dynamic_ir.h:38]: yea you might be right, fell free to send a pr to fix the comment

**Comment by miladm** [comment]
Looking into this PR. Makes sense. Thanks @JackCaoG.

**Pull request merged**